layers/+lang/typescript/funcs.el

```diff
@@ -65,3 +65,19 @@
                  (list (point-min) (point-max))))
   (browse-url (concat "http://www.typescriptlang.org/Playground#src="
                       (url-hexify-string (buffer-substring-no-properties start end)))))
+
+(defun spacemacs//setup-typescript-mode ()
+  ;; Enable js-mode snippets
+  (yas-activate-extra-mode 'js-mode)
+  ;; See https://github.com/syl20bnr/spacemacs/issues/8222
+  (set (make-local-variable 'company-minimum-prefix-length) 2))
+
+(defun spacemacs//setup-typescript-tsx-mode ()
+  ;; Enable js-mode snippets
+  (yas-activate-extra-mode 'js-mode)
+  ;; Force jsx content type
+  (web-mode-set-content-type "jsx")
+  ;; Don't auto-quote attribute values
+  (setq-local web-mode-enable-auto-quoting nil)
+  ;; See https://github.com/syl20bnr/spacemacs/issues/8222
+  (set (make-local-variable 'company-minimum-prefix-length) 2))
```

Comment:
I've never actually observed this issue anywhere, but I have my backends narrowed down to only:

```
(company-tide company-capf
              (company-dabbrev-code company-keywords)
              company-files company-dabbrev)
```

Does this work with `setq-local`? afaik they should be about the same thing and it's weird to have a `setq-local` above and this below.

Without this what is your `company-minimum-prefix-length` set to in a typescript buffer?

Also, after a `{`, what does `company-diag` report?

Reply:
Also, this should be hooked from a post-init-company

Reply:
I get 
```
company-backends: ((company-tide :with company-yasnippet)
 (company-dabbrev-code company-gtags company-etags company-keywords :with company-yasnippet)
 (company-files :with company-yasnippet)
 (company-dabbrev :with company-yasnippet))
```

Will move it in the post-init and see if it works with setq-local

Reply:
Also, after a `{`, what does `M-x company-diag` report?

Reply:
And without this what is your `company-minimum-prefix-length` set to in a typescript buffer?

Reply:
1. extremely long output that starts with the company backends listed above
2. minimum prefix is set to 0

Reply:
I tried
```
(defun typescript/post-init-company ()
  (set (make-local-variable 'company-minimum-prefix-length) 2))
```
or
```
(defun typescript/post-init-company ()
  (setq-local company-minimum-prefix-length 2))
```
or
```
(defun typescript/post-init-company ()
  (setq company-minimum-prefix-length 2))
```
but none seems to work

Reply:
> extremely long output that starts with the company backends listed above

I'm interested in what the "Used backend" is. Mine is "none" at this point.

> minimum prefix is set to 0

Ah. Do you have the html layer enabled by chance? It looks like that's what is setting that to 0 for web-mode because of https://github.com/osv/company-web/issues/4

Reply:
yes, they need to be inside a hook.